pyrules/tests/test_conditions.py

```diff
@@ -0,0 +1,249 @@
+        self.assertEqual(c1._expr, c2._expr)
+
+    def test_combine(self):
+        c1 = C('foo', 'bar')
+        h1 = hash(c1)
+        c1.combine(C('x', True), C.OR)
+        self.assertEqual(hash(c1), h1)
+        self.assertEqual(
+            c1._expr,
+            (C.OR, ('foo', 'bar', False), ('x', True, False), False))
+
+    def test_apply(self):
+        self.assertEqual(C(foo=True).apply({'foo': True}), True)
+        self.assertEqual(C(foo=True).apply({'foo': False}), False)
+        self.assertEqual((~C(foo=True)).apply({'foo': True}), False)
+        self.assertEqual((~C(foo=True)).apply({'foo': False}), True)
+        self.assertEqual(
+            (C(foo=True) & C(bar=True)).apply({'foo': True, 'bar': True}),
+            True)
+        self.assertEqual(
+            (C(foo=True) & C(bar=True)).apply({'foo': False, 'bar': True}),
+            False)
+        self.assertEqual(
+            (C(foo=True) & C(bar=True)).apply({'foo': True, 'bar': False}),
+            False)
+        self.assertEqual(
+            (C(foo=True) & C(bar=True)).apply({'foo': False, 'bar': False}),
+            False)
+        for c1 in (True, False):
+            for c2 in (True, False):
+                for c3 in (True, False):
+                    self.assertEqual(
+                        (~C(c1=True) ^ (C(c2=True) | C(c3=True))).apply(
+                            {'c1': c1, 'c2': c2, 'c3': c3}),
+                        (not c1) ^ (c2 | c3))
+
+    def test_negation(self):
+        self.assertNotEqual(~C('foo', 'bar'), C('foo', 'bar'))
+        self.assertEqual(~(~C('foo', 'bar')), C('foo', 'bar'))
+        
+    def test_simple_cond(self):
+        self.assertEqual(C('foo', 'bar')._expr, ('foo', 'bar', False))
+        self.assertEqual(C(foo=True)._expr, ('foo', True, False))
+        self.assertEqual(C(foo=False)._expr, ('foo', False, False))
+
+    def test_simple_cond_negate(self):
+        self.assertEqual(C('foo', 'bar', True)._expr, ('foo', 'bar', True))
+        self.assertEqual((~C(foo=True))._expr, ('foo', True, True))
+
+    def test_cond_ops(self):
+        self.assertEqual(
+            (C(foo=True) & C(bar=False))._expr,
+            (C.AND, ('foo', True, False), ('bar', False, False), False))
+        self.assertEqual(
+            (C(foo=True) | C(bar=False))._expr,
+            (C.OR, ('foo', True, False), ('bar', False, False), False))
+        self.assertEqual(
+            (C(foo=True) ^ C(bar=False))._expr,
+            (C.XOR, ('foo', True, False), ('bar', False, False), False))
+
+    def test_cond_ops_negate(self):
+        self.assertEqual(
+            (~C(foo=True) & C(bar=False))._expr,
+            (C.AND, ('foo', True, True), ('bar', False, False), False))
+        self.assertEqual(
+            (~(C(foo=True) & C(bar=False)))._expr,
+            (C.AND, ('foo', True, False), ('bar', False, False), True))
+
+    def test_repr(self):
+        self.assertEqual(repr(C(foo='bar')), '<C: foo == bar>')
```

Comment:
I know it works, but please add one more  test case (or test cases):

`c11 = C(foo__contains='bar', foo__eq='bar')`
=> <C: (foo__contains == bar AND foo__eq == bar)>